plugins/org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/IncrementalBuilder.xtend

```diff
@@ -71,9 +77,10 @@ import org.eclipse.xtext.validation.CheckMode
 					if (resource.validate) {
 						resource.generate(request, newSource2GeneratedMapping)
 					}
-					return true
+					val old = request.previousState.resourceDescriptions.getResourceDescription(resource.URI)
+					return new DefaultResourceDescriptionDelta(old, copiedDescription)
```

Comment:
this should probably be IResourceDescription.Manager.createDelta(IResourceDescription, IResourceDescription)